plugin_sets:
  example:
    ### FILTERS ###
    pcl_pass_through_box_filter:
      params:
        field_name: z
        limit_min: -1.0
        limit_max: 1.0

    pcl_pass_through_ellipse_filter:
      params:
        radius_x: 4.0
        radius_y: 2.0

    pcl_voxel_grid_filter:
      params:
        lx: 0.02
        ly: 0.02
        lz: 0.05

    ### Sensors ###
    generic_lidar_sensor:
      params:
        topic: point_cloud_update
        filter_chain:
          - pcl_pass_through_box_filter
          - pcl_pass_through_ellipse_filter
        process_chain:
          - grid_map_merger

    image_map_sensor:
      params:
        topic: raw_image_map
        sensor_frame: map
        resolution: 0.02 # Map resolution based on image pixels [m]
        min_height: 0.0 # Height of the min greyscale color on image (Black: 0;0;0) in [m]
        max_height: 1.0 # Height of the max greyscale color on image (White: 255;255;255) in [m]
        filter_chain:
          - pcl_pass_through_box_filter
        process_chain:
          - grid_map_merger

    ### Generators ###
    grid_map_merger:
      import: grid_map_merger
      params:
        topic: /traversability_estimation/traversability_map
        in_layers: [ traversability, traversability, elevation ]
        out_layers: [ traversability_heuristic, traversability_cost, elevation ]
        apply_smoothing: [ True, True, False ]
        resolution: 0.05
        unknown_area_value: [ 0.9, 0.9, 0.0 ]
        gaussian_kernel_width: [ 7, 5, 1 ]
        gaussian_kernel_height: [ 7, 5, 1 ]
        sigma_x: [ 7.0, 5.0, 1.0 ]
        sigma_y: [ 0.0, 0.0, 0.0 ]
        erosion_kernel_width: [ 1, 1, 1 ]
        erosion_kernel_height: [ 1, 1, 1 ]
        process_chain:
          - grid_map_publisher

    ### Publisher ###
    terrain_model_publisher: none
    grid_map_publisher: none
